photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/GameTrainingDriver.scala

```diff
@@ -0,0 +1,711 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.cli.game.training
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.SparkContext
+import org.apache.spark.ml.param.{Param, ParamMap, ParamValidators, Params}
+import org.apache.spark.sql.DataFrame
+
+import com.linkedin.photon.ml.HyperparameterTuningMode.HyperparameterTuningMode
+import com.linkedin.photon.ml._
+import com.linkedin.photon.ml.TaskType.TaskType
+import com.linkedin.photon.ml.Types._
+import com.linkedin.photon.ml.cli.game.GameDriver
+import com.linkedin.photon.ml.data.{DataValidators, InputColumnsNames}
+import com.linkedin.photon.ml.data.avro.{AvroDataReader, ModelProcessingUtils}
+import com.linkedin.photon.ml.estimators.GameEstimator.GameOptimizationConfiguration
+import com.linkedin.photon.ml.estimators.{GameEstimator, GameEstimatorEvaluationFunction}
+import com.linkedin.photon.ml.hyperparameter.search.{GaussianProcessSearch, RandomSearch}
+import com.linkedin.photon.ml.io.{CoordinateConfiguration, ModelOutputMode}
+import com.linkedin.photon.ml.io.ModelOutputMode.ModelOutputMode
+import com.linkedin.photon.ml.io.scopt.ScoptGameTrainingParametersParser
+import com.linkedin.photon.ml.normalization.NormalizationType.NormalizationType
+import com.linkedin.photon.ml.normalization.{NormalizationContext, NormalizationType}
+import com.linkedin.photon.ml.optimization.game.CoordinateOptimizationConfiguration
+import com.linkedin.photon.ml.stat.BasicStatisticalSummary
+import com.linkedin.photon.ml.util.Implicits._
+import com.linkedin.photon.ml.util.Utils
+import com.linkedin.photon.ml.util._
+
+/**
+ * This object is the entry point and driver for GAME training. There is a separate driver object for scoring.
+ */
+object GameTrainingDriver extends GameDriver {
```

Comment:
@joshvfleming 
@fastier-li 
@Adama94

Aside from the new parameters, this class is mostly identical to the `training.Driver`. Like the `GameDriver` and 'GameEstimator`, this object defines the parameters it expects to use (in this case, for GAME training). Like the `GameEstimator`, it has a `validateParams` function to test interactions between the parameters, and a `setDefaultParams` function to setup the parameter defaults. Further changes are almost all as a consequence of replacing `GameTrainingParams`.